### PR TITLE
(#140) SCOM tests

### PR DESCRIPTION
**Pull request by rexim**
As per #140

`SCOM` value for [OneVoidMethodWithoutParams] and [OneMethodCreatesLambda] cannot be `NaN` for the same reason as in #207. See [SCOM.xsl#L118]. The value is `NaN` when amount of methods is less than 1 or when amount of attributes is 0. [OneVoidMethodWithoutParams] and [OneMethodCreatesLambda] have in fact two methods because the Skeleton treats ctor as a method. See this discussion: https://github.com/yegor256/jpeek/pull/207#issuecomment-367281359

[OneVoidMethodWithoutParams]: https://github.com/yegor256/jpeek/blob/fc4437faaff683cfecb073cb7e04f3418cdefca9/src/test/resources/org/jpeek/samples/OneVoidMethodWithoutParams.java
[OneMethodCreatesLambda]: https://github.com/yegor256/jpeek/blob/fc4437faaff683cfecb073cb7e04f3418cdefca9/src/test/resources/org/jpeek/samples/OneMethodCreatesLambda.java
[SCOM.xsl#L118]: https://github.com/yegor256/jpeek/blob/fc4437faaff683cfecb073cb7e04f3418cdefca9/src/main/resources/org/jpeek/metrics/SCOM.xsl#L118

### CODE REVIEW

**Comment by 0crat** [comment]
Job [#210](https://github.com/yegor256/jpeek/issues/210) is now in scope, role is `REV`

**Comment by 0crat** [comment]
This pull request [#210](https://github.com/yegor256/jpeek/issues/210) is assigned to @amihaiemil[/z](https://www.0crat.com/u/amihaiemil), here is [why](https://www.0crat.com/footprint/C7JGJ00DP/541411543). The budget is 15 minutes, see [§4](http://www.zerocracy.com/policy.html#4). Please, read [§27](http://www.zerocracy.com/policy.html#27) and when you decide to accept the changes, inform @yegor256[/z](https://www.0crat.com/u/yegor256) (the architect) right in this ticket. If you decide that this PR should not be accepted ever, also inform the architect.

File: src/test/java/org/jpeek/MetricsTest.java [hunk 1]
```diff
@@ -129,6 +129,10 @@
             new Object[] {"TwoCommonAttributes", "NHD", 0.3333d},
             new Object[] {"MethodsWithDiffParamTypes", "CCM", 0.0476d},
             new Object[] {"TwoCommonAttributes", "SCOM", 0.0d},
+            new Object[] {"NoMethods", "SCOM", Double.NaN},
+            new Object[] {"OneVoidMethodWithoutParams", "SCOM", 0.0d},
```
Comment (amihaiemil) [inline src/test/java/org/jpeek/MetricsTest.java]: @rexim ok, value for this class and the other won't be ``Nan``, but why is it ``0.0d``, how did you calculate this?

Comment (rexim) [reply src/test/java/org/jpeek/MetricsTest.java]: @amihaiemil According to [the definition of SCOM](https://github.com/yegor256/jpeek/blob/fc4437faaff683cfecb073cb7e04f3418cdefca9/src/main/resources/org/jpeek/metrics/SCOM.xsl#L38):

> "0" indicates no cohesion at all (i.e. every method deals with an independent set of attributes)

Which is absolutly true for both of the samples. Both of them have a single method that deals with its own attributes and a sinlge ctor that doesn't deal with any attributes at all. No cohesion whatsoever. You can take those methods and move them to another classes and the ctors won't notice a thing.

**Comment by rexim** [comment]
@amihaiemil done. Please take a look again when you get a chance.

**Comment by amihaiemil** [comment]
@rultor good to merge

**Comment by amihaiemil** [comment]
@yegor256 can we merge this?

**Comment by yegor256** [comment]
@rultor merge

**Comment by 0crat** [comment]
@ypshenychka[/z](https://www.0crat.com/u/ypshenychka) please review this job, as in [§30](http://www.zerocracy.com/policy.html#30); the job will be fully closed and all payments will be made when the quality review is completed

**Comment by ypshenychka** [comment]
@amihaiemil According to our [QA Rules](http://www.zerocracy.com/policy.html#42):
>The code reviewer found at least three problems in the code.
Comments were mostly about design problems, not cosmetic issues.

No issues were found during code review. 
Please confirm that you'll try to find at least three problems while future reviews.

**Comment by amihaiemil** [comment]
@ypshenychka I confirm. Again, PR is too small for double review :)

**Comment by ypshenychka** [comment]
@amihaiemil Thanks! I see :) Probably policy will be revised soon.

**Comment by ypshenychka** [comment]
@0crat quality acceptable

**Comment by amihaiemil** [comment]
@ypshenychka Can you comment again the verdict? 0crat missed it apparently.

**Comment by 0crat** [comment]
`QA` review completed: +8 points just awarded to @ypshenychka[/z](https://www.0crat.com/u/ypshenychka), total is [+797](https://www.0crat.com/u/ypshenychka)

**Comment by 0crat** [comment]
Order was finished, quality was "acceptable": +15 points just awarded to @amihaiemil[/z](https://www.0crat.com/u/amihaiemil), total is [+1085](https://www.0crat.com/u/amihaiemil)

**Comment by 0crat** [comment]
> @0crat quality acceptable ([here](https://github.com/yegor256/jpeek/issues/210#issuecomment-368576780))

@ypshenychka The job [#210](https://github.com/yegor256/jpeek/issues/210) is now out of scope

**Comment by 0crat** [comment]
Payment to `ARC` for a closed pull request, as in [§28](http://www.zerocracy.com/policy.html#28): +10 points just awarded to @yegor256[/z](https://www.0crat.com/u/yegor256), total is [+10964](https://www.0crat.com/u/yegor256)

**Pull request merged**